From 32e1ca9cff47560d68a69b94cfc3ee9f1f256080 Mon Sep 17 00:00:00 2001
From: Thomas Renninger <trenn@suse.com>
Date: Wed, 24 Jan 2024 11:43:43 +0100
Subject: [PATCH 2/4] Make use of standard variables for installation

otherwise %cmake rpm macros will not be able to install things to
the right places.

Signed-off-by:  <trenn@suse.com>
---
 CMakeLists.txt | 18 ++++++++++++++----
 1 file changed, 14 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 71dcf46..80943e4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,6 +14,16 @@ set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/bin)
 set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
 set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
 
+if(NOT DEFINED INCLUDE_INSTALL_DIR)
+  set(INCLUDE_INSTALL_DIR include)
+endif()
+if(NOT DEFINED CMAKE_INSTALL_BINDIR)
+  set(CMAKE_INSTALL_BINDIR bin)
+endif()
+if(NOT DEFINED CMAKE_INSTALL_LIBDIR)
+  set(CMAKE_INSTALL_LIBDIR lib)
+endif()
+
 set(CMAKE_C_FLAGS_DEBUG "-D_DEBUG")
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}")
@@ -92,10 +102,10 @@ if(CZMQ_FOUND)
 endif()
 
 install(TARGETS redfishtest redfish
-        RUNTIME DESTINATION bin
-        LIBRARY DESTINATION lib)
-install(FILES ${REDFISH_HDR_PUBLIC_RED} DESTINATION include)
-install(FILES ${REDFISH_HDR_PUBLIC_ENTITIES} DESTINATION include/entities)
+        RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+        LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+install(FILES ${REDFISH_HDR_PUBLIC_RED} DESTINATION "${INCLUDE_INSTALL_DIR}")
+install(FILES ${REDFISH_HDR_PUBLIC_ENTITIES} DESTINATION "${INCLUDE_INSTALL_DIR}/entities")
 
 ENABLE_TESTING()
 
-- 
2.43.0

